gigascience/gigadb-website
# reminting notification text change

# Pull request for issue: #1830

This is a pull request for the following functionalities:

* refactored frontend code of minting button
* restyled alert to align to inputs
* updated alert message as indicated in the issue

## How to test?

* You can see in the code that when DOI and metadata exist and metadata is updated, the updated message is provided, This use case seems difficult to reproduce in local dev without hardcoding a mock response

## How have functionalities been implemented?

* the code in charge of handling the minting response was hard to read and lacked (for me) syntax highlight so I separated it into its own script block

## Any issues with implementation?

* In local dev, the response from the post request is either unhandled `null` or a case not handled by the conditional branches. I do not know whether this is intended behavior or something that only happens in local dev and works as intended in production.
  * for the unhandled cases I added a fallback "Unexpected error". I suspect this should never trigger, but in case it triggers, at least something is displayed. I see 5 use cases covered in the frontend logic (`protected/views/adminDataset/_form.php`), none of the cases catch the local dev behavior. Are we sure that the current use cases cover all production cases?

## Any changes to automated tests?

- the tests related to the doi require gitlab secret. I replaced the text in the suitable acceptance test
